sql/dd/impl/upgrade/dd.cc

```diff
@@ -1031,6 +1032,84 @@ bool update_object_ids(THD *thd, const std::set<String_type> &create_set,
 }
 /* purecov: end */
 
+/*
+  upgrade dd_properties table during dd_engine change
+
+  @param  thd                      Thread context.
+  @param  mysql_schema_id          Id of the 'mysql' schema.
+  @param  target_table_schema_id   Id of the schema where the tables in the
+                                   create_set are located.
+  @param  target_table_schema_name Name of the schema where the tables in the
+                                   create_set are located.
+  @param  actual_table_schema_id   Id of the schema where the tables in the
+                                   remove_set will be moved.
+
+  @returns false if success, true otherwise.
+
+*/
+bool upgrade_dd_properties_table(THD *thd, Object_id mysql_schema_id,
+                                 Object_id target_table_schema_id,
+                                 const String_type &target_table_schema_name,
+                                 Object_id actual_table_schema_id) {
+  // only upgrade dd_properties table for dd_engine table
+  // during dd_version change, dd_properties table keep same
+  if (!bootstrap::DD_bootstrap_ctx::instance().is_dd_engine_change())
+    return false;
+
+  // create <upgrade>.dd_properties table with default_ddse
+  dd::tables::DD_properties::instance().set_actual_engine(get_dd_engine_name());
```

Comment:
The code uses mixture of `auto` and explicit types, i.e. handlerton type variables tend to be explicit and handler ones auto. Please consider using a single style, preferably `auto`